src/tools/demonstrate_amplitude_tool.rb

```diff
@@ -0,0 +1,94 @@
+require_relative 'spring_simulation_tool.rb'
+require 'src/system_simulation/trace_visualization.rb'
+
+# Enables users to drag a line starting from a node to demonstrate the amplitude they want for the oscillation.
+class DemonstrateAmplitudeTool < SpringSimulationTool
+  def initialize(ui)
+    super(ui)
+    @mouse_input = MouseInput.new(snap_to_edges: true, snap_to_nodes: true)
+
+    @start_node = nil
+    @start_position = nil
+    @end_position = nil
+    @moving = false
+
+    @trace_visualization = TraceVisualization.new
+  end
+
+  def onLButtonDown(_flags, x, y, view)
+    @mouse_input.update_positions(view, x, y)
+    obj = @mouse_input.snapped_object
+    if !obj.nil? && obj.is_a?(Node)
```

Comment:
Guard clause instead of wrapping the code inside is actually a good idea I think